pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -281,11 +288,19 @@ func (ds *dockerService) getIP(sandbox *dockertypes.ContainerJSON) (string, erro
 		// reporting the IP.
 		return "", nil
 	}
-	if IP, err := ds.getIPFromPlugin(sandbox); err != nil {
-		glog.Warningf("%v", err)
-	} else if IP != "" {
-		return IP, nil
+
+	// Don't bother getting IP if networking isn't ready
```

Comment:
If kubelet restarts, it'd lose the in-memory cache and not get the IP for any sandbox.
It's hard to eliminate the race solely based on the sandbox ID since we don't get it until the underlying container is created. An alternative is to use the sandbox name as a key in the map.
 - Add the key to the map *before* creating the container.
 - If the key exists, assume the network is being set up, and don't check the IP.
 - If the key does not exist, get the IP as usual. 
 - Delete the key once the setup is done.